cpp/modmesh/buffer/pymod/wrap_SimpleArray.cpp

```diff
@@ -407,10 +407,136 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
+            return py::cast(WrapSimpleArray<int16_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "int32")
+            return py::cast(WrapSimpleArray<int32_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "int64")
+            return py::cast(WrapSimpleArray<int64_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "uint8")
+            return py::cast(WrapSimpleArray<uint8_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "uint16")
+            return py::cast(WrapSimpleArray<uint16_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "uint32")
+            return py::cast(WrapSimpleArray<uint32_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "uint64")
+            return py::cast(WrapSimpleArray<uint64_t>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "float32")
+            return py::cast(WrapSimpleArray<float>::init_shape(args[0].cast<py::tuple>()));
+        if (type == "float64")
+            return py::cast(WrapSimpleArray<double>::init_shape(args[0].cast<py::tuple>()));
+    }
+    else if (arg_len == 3 && py::isinstance<py::tuple>(args[0]))
+    {
+        if (type == "bool")
+            return py::cast(WrapSimpleArray<bool>::init_value(args[0].cast<py::tuple>(), args[1].cast<bool>()));
+        if (type == "int8")
+            return py::cast(WrapSimpleArray<int8_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<int8_t>()));
+        if (type == "int16")
+            return py::cast(WrapSimpleArray<int16_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<int16_t>()));
+        if (type == "int32")
+            return py::cast(WrapSimpleArray<int32_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<int32_t>()));
+        if (type == "int64")
+            return py::cast(WrapSimpleArray<int64_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<int64_t>()));
+        if (type == "uint8")
+            return py::cast(WrapSimpleArray<uint8_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<uint8_t>()));
+        if (type == "uint16")
+            return py::cast(WrapSimpleArray<uint16_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<uint16_t>()));
+        if (type == "uint32")
+            return py::cast(WrapSimpleArray<uint32_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<uint32_t>()));
+        if (type == "uint64")
+            return py::cast(WrapSimpleArray<uint64_t>::init_value(args[0].cast<py::tuple>(), args[1].cast<uint64_t>()));
+        if (type == "float32")
+            return py::cast(WrapSimpleArray<float>::init_value(args[0].cast<py::tuple>(), args[1].cast<float>()));
+        if (type == "float64")
+            return py::cast(WrapSimpleArray<double>::init_value(args[0].cast<py::tuple>(), args[1].cast<double>()));
+    }
+    throw std::runtime_error("unsupported type.");
+}
+
 void wrap_SimpleArray(pybind11::module & mod)
 {
+    mod.def("SimpleArray", &createSimpleArray);
```

Comment:
This is not a class (type), but a helper function using the class naming convention.  The returned type depends on the input argument, e.g.,

```python
type(SimpleArray((2, 3), dtype='int32'))) != type(SimpleArray((2, 3), dtype='float32')))
```

But we want the are the same type.  Thus, in the pybind11 wrapper, there should be a Python type created as a facade.